java/src/test/java/org/rocksdb/util/BytewiseComparatorTest.java

```diff
@@ -242,14 +242,22 @@ private void doRandomIterationTest(
             result_iter.seekToLast();
             break;
           case 2: {
-            // Seek to random key
+            // Seek to random (existing or non-existing) key
             final int key_idx = rnd.nextInt(source_strings.size());
-            final String key = source_strings.get(key_idx);
+            final String key = source_strings.get(key_idx) + (rnd.nextBoolean() ? "" : "non-existing");
```

Comment:
The previous version of testing the `seek()`-case had selected its random keys only via `source_strings.get(key_idx)`, which is passed as an argument of value `("a", "b", "c", "d", "e", "f", "g", "h", "i")`. Thus, it queried only for *existing* keys, although the main purpose of `seek()` is to query for non-existing keys that may lay in between our outside the range of existing keys (rounded up by `seek()` or rounded down by `seekForPrev()`).
Thus, a test with better coverage would, for example, base on adding only the keys `("b", "d", "f", "h")` to the database, and then query (in case `seek()` and `seekForPrev()`) by random for any of `("a", "b", "c", "d", "e", "f", "g", "h", "i")`. This would check for correct rounding up/down.
In order to "simulate" this extended test without sneaking the knowledge about the argument value into the function, I added the random modification `+ (rnd.nextBoolean() ? "" : "non-existing")`. This appends by random either the empty string or the string `"non-existing"` to the probed key. So it queries in half of the cases for an existing key (e.g., `"b"`), and in the other half of the cases for some obviously non-existing key (e.g., `"bnon-existing"`) that lies in between the modified key and the next one (if present). However, this compromise solution cannot generate a modified key *below* `"a"`, so test coverage is still not perfect...
Summarized, the purpose of this modification is to check whether the up/down-rounding behaviour  of rocksdb is the same as of the simulated ground truth.
To me, any of the following 3 solutions is fine:

1. leave the modification in as it is
1. modify the test as explained above using a modified argument to the test function
1. remove the modification, but add a separate new test that is only dedicated to the rounding-behaviour of `seek()` and `seekForPrev()`

Which solution do you prefer?